ci/common.groovy

```diff
@@ -0,0 +1,60 @@
+def installJSDeps(platform) {
+  def attempt = 1
+  def maxAttempts = 10
+  def installed = false
+  /* prepare environment for specific platform build */
+  sh "scripts/prepare-for-platform.sh ${platform}"
+  while (!installed && attempt <= maxAttempts) {
+    println "#${attempt} attempt to install npm deps"
+    sh 'npm install'
+    installed = fileExists('node_modules/web3/index.js')
```

Comment:
This is fragile. Usual case with npm is that it fails to install **all** dependencies. I would just check `npm install` exit code.

Reply:
I mentioned it in the other comment but `npm ls` also return non-zero exit code if the dependencies differ between what is installed and `package-lock.json`.

Reply:
I didn't write it, and did not want to change it here. This PR aleady changes a fuckton of things.